nan.h

```diff
@@ -1402,7 +1402,7 @@ class NanCallback {
     return NanNew(handle)->Get(kCallbackIndex)->IsUndefined();
   }
 
-  v8::Handle<v8::Value> Call(int argc, v8::Handle<v8::Value> argv[]) const {
+  v8::Handle<v8::Value> Call(int argc, v8::Handle<v8::Value> argv[], v8::Handle<v8::Object> target=NanGetCurrentContext()->Global()) const {
```

Comment:
It's not wrong but the arguments are in the wrong order when compared to v8::Function::Call() or node::MakeCallback().  Why don't you overload Call() like this?

```c++
v8::Handle<v8::Value> Call(v8::Handle<v8::Object> recv, int argc, v8::Handle<v8::Value> argv[]) const {
  // ...
}

v8::Handle<v8::Value> Call(int argc, v8::Handle<v8::Value> argv[]) const {
  return Call(NanGetCurrentContext()->Global(), argc, argv);
}
```

(There's something of a lost performance opportunity there because NanGetCurrentContext() always retrieves the current isolate from thread-local storage.)

Aside: newer versions of V8 let you call a function with a non-object as the receiver.  I don't believe node::MakeCallback() currently reflects that, though.